cloudsync/event.py

```diff
@@ -86,10 +87,10 @@ def forget(self):
 
     @property
     def busy(self):
-        return not self.events.empty or self.need_walk
+        self.do()
```

Comment:
why are we doing this? This seems like a lot of work to be hidden in a property

Reply:
true, though as of now this property is only used in tests AFAIK -- e.g., 

sync.run(until=lambda: not sync.busy, timeout=1)

i'm open to suggestions as to how this could be done better, given that we are dealing with generators of events at the provider level.

Reply:
unfortunately, "busy" needs to work this way.   feel free to change to a method "is_busy()" 
 and/or change all the tests to use the method instead, etc.   still going to be heavy for an is_check.   fortunately there's actually nothing very heavy done in the event manager.

Reply:
"is_busy()" can be changed to stuff pending events into a queue instead of actually processing them, and then return false if the queue is empty.  (to make it a bit more lightweight)

Reply:
thinking we can leave this as is for now, perhaps with a todo note to come back to it later -- renaming to is_busy() doesn't really change anything, the queue would help but would add some complexity (this is similar to what the muxer was doing)

any strong objections?